lib/evaluate/int-power.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_EVALUATE_INT_POWER_H_
+#define FORTRAN_EVALUATE_INT_POWER_H_
+
+// Computes an integer power of a real or complex value.
+
+#include "common.h"
+
+namespace Fortran::evaluate {
+
+template<typename REAL, typename INT>
+ValueWithRealFlags<REAL> IntPower(const REAL &base, const INT &power,
+    Rounding rounding = Rounding::TiesToEven) {
+  REAL one{REAL::FromInteger(INT{1}).value};
+  ValueWithRealFlags<REAL> result;
+  result.value = one;
+  if (base.IsNotANumber()) {
```

Comment:
We seem to be using "NaN" everywhere else, so why not "IsNaN" for this function?
(I know this wasn't added in this change -- sorry.)

Reply:
I'll rename some of the others to avoid the needless use of an obscure term of art.